### ISSUE TITLE
Parsing issue with arrow functions with type specs that returns JSX 

### ISSUE THREAD

**Issue by etrepum** [prettier/prettier#2286]
This is a regression in 1.5.0.

Here's a minimal repro of the parsing issue:

https://prettier.io/#%7B%22content%22%3A%22const%20pass%20%3D%20()%3A%20X%20%3D%3E%20(%3Cx%20%2F%3E)%3B%5Cnconst%20fail%20%3D%20()%3A%20X%20%3D%3E%20%3Cx%20%2F%3E%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D

```javascript
const pass = (): X => (<x />);
const fail = (): X => <x />;
```

```
SyntaxError: Unexpected token, expected , (2:26)
```

Note that if you comment out `fail`, the first run will succeed, but it generates equivalent code to `fail` so a subsequent pass will not succeed.

**Comment by etrepum** [comment]
On further inspection, this works with the flow parser, but not the default babylon. In 1.4.x this worked with the default options.

**Comment by vjeux** [comment]
Thanks for the really good repro. I'm investigating inside of babylon to see what's going on.

**Comment by vjeux** [comment]
cc @existentialism, @hzoo, @nicolo-ribaudo

**Comment by existentialism** [comment]
Looks like it regressed in https://github.com/babel/babylon/commit/6b4fba4deb9d4e2261c0514db6d45e068eb53592.

**Comment by vjeux** [comment]
I bisected in prettier to: Bump babylon version (#2205) fwiw.

**Comment by vjeux** [comment]
@existentialism I'd love to do a hotfix for prettier soon (tens of minutes) to fix this. Should I revert #2205, or can you send a fix to babylon and I can use a specific commit for it?

**Comment by existentialism** [comment]
Could just pin to b13 and remove the func expr in jsx attr test for now?

**Comment by vjeux** [comment]
Sounds good. Thanks!

**Comment by vjeux** [comment]
I just published 1.5.1 which fixes it, thanks for raising it!

**Comment by nicolo-ribaudo** [comment]
@vjeux This issue will be fixed in the next Babylon release! (babel/babylon#595)
